dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerStructureFinderStrategyResolver.java

```diff
@@ -163,33 +208,74 @@ public List<ContainerStructure> apply(final Container container) {
                         new ImmutableList.Builder<>();
 
                 try {
-
-                    final HibernateUtil dh = new HibernateUtil(ContainerStructure.class);
-                    dh.setSQLQuery("select {container_structures.*} from container_structures " +
-                            "where container_structures.container_id = ? " +
-                            "and container_structures.container_inode = ?");
-                    dh.setParam(container.getIdentifier());
-                    dh.setParam(container.getInode());
-                    builder.addAll(dh.list());
-
-                    //Add the list to cache.
-                    containerStructures = builder.build();
-                    containerStructures = containerStructures.stream().map((cs)-> {
-                        if(cs.getCode()==null) {
-                            cs.setCode(""); 
-                        }
-                        return cs;
-                    }).collect(Collectors.toList());
+                    if (Container.SYSTEM_CONTAINER.equals(container.getIdentifier())) {
+                        addAllContentTypesToSystemContainer(builder);
+                        containerStructures = builder.build();
+                    } else {
+                        final HibernateUtil dh = new HibernateUtil(ContainerStructure.class);
```

Comment:
doing this is a violation layers, this should be in a API + Factory
in the worst case, encapsulate this in method and annotated with Close db if opened

Reply:
I did not add that code in my PR, but I can take of changing it.

Reply:
**NOTE: This change will be handled in another branch**